appinventor/appengine/src/com/google/appinventor/client/editor/simple/SimpleNonVisibleComponentsPanel.java

```diff
@@ -18,11 +19,14 @@
 import com.google.gwt.user.client.ui.VerticalPanel;
 import com.google.gwt.user.client.ui.Widget;
 
+import java.util.List;
+import java.util.Map;
+
 /**
  * Panel in the Simple design editor holding non-visible Simple components.
  *
  */
-public final class SimpleNonVisibleComponentsPanel extends Composite implements DropTarget {
+public final class SimpleNonVisibleComponentsPanel extends Composite implements DropTarget, ComponentDatabaseChangeListener {
```

Comment:
The ComponentDatabaseChangeListener implementation does nothing, but it serves as an example for Panels that could change upon a Component Upgrade.

The SimpleNonVisibleComponentsPanel does not require this now. But who knows it wont in the future..